src/main/scala/io/hydrosphere/mist/master/MasterServer.scala

```diff
@@ -39,78 +39,89 @@ class MasterServer(
-      name = Constants.Actors.cliResponderName)
-
-    logger.info(s"${Constants.Actors.cliResponderName}: started")
-
-    masterService.recoverJobs()
-
-    httpBinding = Some(bootstrapHttp(streamer, masterService))
+    for {
+      logService       <- LogStreams.runService(
+                            config.logs.host, config.logs.port,
+                            logsMappings, eventPublishers)
+      _                =  {
+                             logger.info(s"Logger Service is started")
+                             logsService = Some(logService)
+                       }
+      jobsLogger       =  logService.getLogger
+      statusService    =  system.actorOf(StatusService.props(store, eventPublishers, jobsLogger), "status-service")
+      workerManagerVal =  system.actorOf(
+                            WorkersManager.props(
+                              statusService, workerRunner,
+                              jobsLogger,
+                              config.workers.runnerInitTimeout,
+                              infoProvider
+                            ), "workers-manager")
+      _                =  {
+                             logger.info("Worker Manager is Started")
+                             workerManager = Some(workerManagerVal)
+                       }
+      contexts         <- contextsStorage.precreated
+      _                =  contexts.foreach(context => {
+                            logger.info(s"Precreate context for ${context.name}")
+                            workerManagerVal ! CreateContext(context)
+                          })
+      jobService       =  new JobService(workerManagerVal, statusService)
+      masterService    =  new MasterService(jobService, endpointsStorage, contextsStorage, logsMappings)
+      _                =  {
+                            system.actorOf(
+                              CliResponder.props(masterService, workerManagerVal),
+                              name = Constants.Actors.cliResponderName)
+                            // Start CLI
+                            logger.info(s"${Constants.Actors.cliResponderName}: started")
+                       }
+      _                <- {
+                            logger.info("Recovering jobs")
+                            masterService.recoverJobs()
+                       }
+      binding          <- {
+                            logger.info(s"Binding http ${config.http.host}:${config.http.port}")
+                            bootstrapHttp(streamer, masterService)
+                       }
+      _                =  {
+                            logger.info(s"Rest started on: $binding")
+                            httpBinding = Some(binding)
+                       }
+      _                =  {
+                            mqttInterface  = bootstrapMqtt(masterService)
+                            kafkaInterface = bootstrapKafka(masterService)
+                       }
+    } yield ()
```

Comment:
Usage of for-comprehension here looks terrible for me.
The better approach here would be to put all steps in `Seq[() -> Future[_]]` and run them one by one.